### PR TITLE
Fixed #703: improve version control of .po file

### LINKED ISSUE #703: Discussion: versionning of po files
As one of developers, I find po files cumbersome when working with pull requests. Every time we run `manage.py makemessages`, Django updates the strings, locations and timestamp of every po file. Locations in particular -- a newline added somewhere in the code will trigger the update of the whole po file. It was sometimes a trouble when I tried to rebase my pull request or merge an other pull request, because it is so common to have various conflicts concerning po files and it takes too long time to resolve them.

Therefore, I would like to open a discussion on the versionning of po files for Sous-Chef. The basic idea comes from the GNU article [Put PO Files under Version Control](https://www.gnu.org/software/gettext/manual/html_node/Translations-under-Version-Control.html) that we may ignore the update of timestamp and locations in everyday development, so that we can make a better review on pull requests. When we prepare a release of Sous-Chef, however, we have to update the whole file in order to help translators on Transifex. How do you think of the idea? Please let me know.

### PR DESCRIPTION
**Pull request by lingxiaoyang**
### Changes proposed in this pull request:

1. Drop English .po files.
2. Remove line numbers from the location strings. This is achieved by extending `django-admin makemessages` command, before an official solution is provided.
3. Disable the line wrap of translatable strings to improve file readability.
4. Replace {% blocktrans %} with {% blocktrans trimmed %} to avoid newline confusion.
5. Provide an example of translator comment.
6. Update developer instructions in i18n doc.
7. Update CONTRIBUTING.md

### Status

- [x] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verify this change

1. See if the doc update makes sense (or your comments on .po file versionning).
2. Test `python3 src/manage.py makemessages -l fr --no-wrap --add-location=file`. See if it works fine.
3. Check updated `django.po` files and see if there are any errors. Don't bother with `fuzzy` ones as they will be double-checked on Transifex.

### Deployment notes and migration

none

### New translatable strings

- [x] If applicable, I have included updated .po files with new strings (see https://github.com/savoirfairelinux/sous-chef/blob/dev/docs/internationalization.adoc)

### Additional notes

none

### CODE REVIEW

File: docs/internationalization.adoc [hunk 1]
```diff
@@ -1,28 +1,57 @@
 ## Instructions for internationalization
```
Comment (lingxiaoyang) [inline docs/internationalization.adoc]: Note: to review this file, please copy and paste it to a gist. The code blocks are not rendered correctly here in the review interface.

File: docs/internationalization.adoc [hunk 2]
```diff
@@ -1,28 +1,57 @@
...
 If commits in your pull request have introduced new translatable content, you must update the .po files and include them in your pull request.
 
 1. `docker exec -it souschef_web_1 bash`
-3. `cd src`
-4. `python3 manage.py makemessages -l fr -l en` to update the various django.po
-files with your new strings.
-5. Commit the changes to your branch to be included in the pull request. Commit only the `django.po` files related to the module that you have worked on, even if Django has updated the `django.po` files in other modules as well.
+2. `cd src`
+3. `python3 manage.py makemessages -l fr --no-wrap --add-location=file` to update the various django.po files with your new strings.
+4. Commit the changes to your branch to be included in the pull request. Commit only the `django.po` files related to the module that you have worked on, even if Django has updated the `django.po` files in other modules as well.
```
Comment (aaboffill) [inline docs/internationalization.adoc]: @lingxiaoyang maybe you can include some text explaining that we can execute  `makemessages`  in the specific package where the new strings was appended, avoiding several not relevant changes. 

`cd package_name`
`python3 ../manage.py makemessages -l fr --no-wrap --add-location=file`

NOTE: I included `../manage.py` instead of `manage.py`, it's a little difference.

With that variation we can generate the `.po` changes only in the specific package changed. 
As an information, could be util to include it.

File: src/billing/views.py [hunk 3]
```diff
@@ -57,8 +57,8 @@ def get(self, request):
 
         messages.add_message(
             self.request, messages.SUCCESS,
-            _("The billing with the identifier #%s \
-            has been successfully created." % billing.id)
+            _("The billing with the identifier #%s "
+              "has been successfully created." % billing.id)
```
Comment (aaboffill) [inline src/billing/views.py]: @lingxiaoyang What do you thin about increase the limit of the code line... we use pep8 and the limit is 79, my appreciation it's so small, we need to spit a lot of code lines. Django use more than 79, exactly [119](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

File: docs/internationalization.adoc (hunk 2 above)
Comment (lingxiaoyang) [inline docs/internationalization.adoc]: @aaboffill  I think it makes sense. I'll add it.

File: src/billing/views.py (hunk 3 above)
Comment (lingxiaoyang) [inline src/billing/views.py]: @aaboffill  I agree. Let's do it in another PR.

**Comment by lingxiaoyang** [comment]
@aaboffill 

> why remove English .po files??

We are using English in the code...

And also, in English .po files, all strings are empty. I don't think they'll be compiled in a later step with "manage.py compilemessages".

File: src/billing/views.py (hunk 3 above)
Comment (aaboffill) [inline src/billing/views.py]: Nice!! @lingxiaoyang , i'll be happy :+1:

**Comment by aaboffill** [comment]
> We are using English in the code...
>
>And also, in English .po files, all strings are empty. I don't think they'll be compiled in a later step with >"manage.py compilemessages".

@lingxiaoyang let me think about that, maybe there are a good reason to keep the English .po files.

**Comment by aaboffill** [comment]
@lingxiaoyang in our methodology of work on `sous-chef` we write always the translatable message in English, your proposal to remove the English `.po` files sound reasonable for me too... go ahead!!!

**Comment by lingxiaoyang** [comment]
@aaboffill  I think I may merge this, based on your approval and the support of @erozqba in the issue discussion #703?

File: docs/internationalization.adoc (hunk 2 above)
Comment (lingxiaoyang) [inline docs/internationalization.adoc]: @aaboffill  I found a problem of this approach. In generated po files, the package directories are dropped:

```
-#: member/templates/client/create/form.html member/tests.py
+#: templates/client/create/form.html tests.py
 #, fuzzy
 #| msgid "emergency_contacts"
 msgid "emergency_contact"
 msgstr "contacts_d'urgence"
 
-#: member/templates/client/create/form.html
-#: member/templates/client/partials/menu.html
-#: member/templates/client/update/base.html
+#: templates/client/create/form.html templates/client/partials/menu.html
+#: templates/client/update/base.html
 msgid "Personal"
 msgstr "Personnel"
```
